sdk/formrecognizer/azure-ai-formrecognizer/samples/v3.2_and_later/async_samples/sample_send_request_async.py

```diff
@@ -43,7 +43,82 @@ async def sample_send_request():
+    )
+    
+    # override the api version to v2.1
+    request = HttpRequest(method="GET", url="v2.1/custom/models?op=summary&api-version=v2.1")
+    async with client:
+        response = client.send_request(request)
+    response.raise_for_status()
+    response_body = response.json()
+    print(f"Our account has {response_body['summary']['count']} custom models, "
+          f"and we can have at most {response_body['summary']['limit']} custom models.")
     
+    # pass with full url and override the api version to v2.1
+    request = HttpRequest(method="GET", url=f"{client._endpoint}/formrecognizer/v2.1/custom/models?op=summary&api-version=v2.1")
+    async with client:
+        response = client.send_request(request)
+    response.raise_for_status()
+    response_body = response.json()
+    print(f"Our account has {response_body['summary']['count']} custom models, "
+          f"and we can have at most {response_body['summary']['limit']} custom models.")
+
+
+async def sample_send_request_v2():
+    endpoint = os.environ["AZURE_FORM_RECOGNIZER_ENDPOINT"]
+    key = os.environ["AZURE_FORM_RECOGNIZER_KEY"]
+
+    client = FormTrainingClient(
+        endpoint=endpoint, credential=AzureKeyCredential(key)
+    )
+    
+    # The `send_request` method can send custom HTTP requests that share the client's existing pipeline,
+    # while adding convenience for endpoint construction and service API versioning.
+    # Now let's use the `send_request` method to make a resource details fetching request.
+    # The URL of the request can be relative (your endpoint is the default base URL),
+    # and the API version of your client will automatically be used for the request.
+    request = HttpRequest(method="GET", url="custom/models?op=summary")
+    async with client:
+        response = client.send_request(request)
+    response.raise_for_status()
+    response_body = response.json()
+    print(f"Our account has {response_body['summary']['count']} custom models, "
+          f"and we can have at most {response_body['summary']['limit']} custom models.")
+    
+    # pass with full url and override the api version
+    request = HttpRequest(method="GET", url=f"{endpoint}/formrecognizer/v2.0/custom/models?op=summary&api-version=v2.0")
+    async with client:
+        response = client.send_request(request)
+    response.raise_for_status()
+    response_body = response.json()
+    print(f"Our account has {response_body['summary']['count']} custom models, "
+          f"and we can have at most {response_body['summary']['limit']} custom models.")
+    
+    # override the api version to 2023-07-31
+    # can only override the api version to 2022-08-31 or later with full url
+    request = HttpRequest(method="GET", url=f"{client._endpoint}/formrecognizer/info?api-version=2023-07-31")
```

Comment:
This is a little confusing to me. The `FormTrainingClient` says that it should only be used on `v2.x`: https://github.com/Azure/azure-sdk-for-python/blob/c8937aeaed4a668ea1e0aebeff4f81368a7f7ce9/sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_form_training_client.py#L43-L44

Does that mean that requests with later API versions are expected to fail? Are we sure that the request being sent here isn't ultimately targeting `v2.1` after the request gets formatted in the generated client?

Reply:
Yes, it will fail. Some operations requires API version to be 2022-08-31 and later. I haven't found an operation works on all API versions. That's why I said it's not necessary to update base url for clients constructed with v2 version, the request will fail anyway.

Reply:
If later API versions fail for this operation, then isn't this snippet proof that you can't override a `v2.1` client to use `2023-07-31` for a request? If sending this request succeeds, that must mean we're targeting the old API version instead of the one specified in the request we're constructing here.

Reply:
No, passing full url works.